tests/general/FilenameBearTest.py

```diff
@@ -118,3 +119,50 @@ def test_auto_file_naming_convention_warning(self):
         self.assertEqual(log.message,
                          'The file naming convention could not be guessed. '
                          'Using the default "snake" naming convention.')
+
+    def test_file_prefix(self):
+        self.section['filename_prefix'] = 'pre'
+        self.check_validity(
+            self.uut, [''], filename='filename.xyz', valid=False)
+        self.check_validity(
+            self.uut, [''], filename='prefilename.xyz', valid=True)
+        self.check_results(
+            self.uut,
+            [''],
+            [Result.from_values('FilenameBear',
+                                'Filename does not use '
+                                'the prefix {!r}.'.format('pre'),
+                                severity=RESULT_SEVERITY.NORMAL,
+                                file='filename.xyz')],
+            filename='filename.xyz')
+
+    def test_file_suffix(self):
+        self.section['filename_suffix'] = 'fix'
+        self.check_validity(
+            self.uut, [''], filename='filename.xyz', valid=False)
+        self.check_validity(
+            self.uut, [''], filename='filenamesuffix.xyz', valid=True)
+        self.check_results(
+            self.uut,
+            [''],
+            [Result.from_values('FilenameBear',
+                                'Filename does not use '
+                                'the suffix {!r}.'.format('fix'),
+                                severity=RESULT_SEVERITY.NORMAL,
+                                file='filename.xyz')],
+            filename='filename.xyz')
+
+    def test_file_prefix_suffix(self):
+        self.section['filename_prefix'] = 'pre'
+        self.section['filename_suffix'] = 'fix'
+        self.check_results(
+            self.uut,
+            [''],
+            [Result.from_values('FilenameBear',
+                                '\n'.join(['- Filename does not use '
+                                           'the prefix {!r}.'.format('pre'),
+                                           '- Filename does not use '
+                                           'the suffix {!r}.'.format('fix')]),
```

Comment:
instead of copying the functionality with join in the algorithm, try to write the string out:
```
"- Filename does not use the prefix 'pre'\n"
"- Filename does not use the suffix 'fix'"
```
So we directly see what we shall get :)
You could do the same above for the `.format` function uses above in the test.